### PR TITLE
fix(risedev): don't kill tmux session unnecessarily

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -720,23 +720,20 @@ if ! ${TMUX} ls &>/dev/null ; then
...
 if [[ -n $(${TMUX} list-windows | grep kafka) ]];
 then
   echo "kill kafka"
   kill_kafka || true
 
   echo "kill zookeeper"
   kill_zookeeper || true
-
-  # Kill their ${TMUX} sessions
-  ${TMUX} list-windows -t risedev -F "#{pane_id}" | xargs -I {} ${TMUX} send-keys -t {} C-c C-d
```
Comment (xxchan) [inline Makefile.toml:739]: This command will kill all windows, and causing `${TMUX} kill-server` below fail with `no server running on /private/tmp/tmux-501/risedev`

Comment (xxchan) [reply Makefile.toml:739]: IIUC, `send-keys C-c C-d` first time will kill the running server, and the second time will kill the tmux window. When all the windows are killed, the tmux server is terminated.

**Review by BugenZhao — COMMENTED** [review]
Are the changes specific to Kafka?

**Review by stdrc — COMMENTED** [review]
Maybe also need to change https://github.com/risingwavelabs/risingwave/blob/63b5954afa52dfd102116f092255862eda7a9dd6/backwards-compat-tests/scripts/utils.sh#L71-L105

File: Makefile.toml [hunk 2]
```diff
@@ -720,23 +720,20 @@ if ! ${TMUX} ls &>/dev/null ; then
...
 fi
 
 ${TMUX} kill-server
```
Comment (kwannoel) [inline Makefile.toml:739]: We can do a post-check here to see if kafka / zookeeper / rw nodes are still running. If yes, we can exit 1.

**Review by kwannoel — APPROVED** [review]

**Review by BugenZhao — APPROVED** [review]
I think that `kafka` and `zookeeper` will soon be replaced by docker-backed services, allowing us to use `docker stop` directly.

**Review by stdrc — DISMISSED** [review]
Maybe also need to change https://github.com/risingwavelabs/risingwave/blob/9996e758cb7a05c931053e8ff7ca3a2094dfaf9a/backwards-compat-tests/scripts/utils.sh#L71-L104

**Comment by xxchan** [comment]
> Maybe also need to change

Whyd o you send twice

File: Makefile.toml (hunk 2 above)
Comment (xxchan) [reply Makefile.toml:746]: I guess we can use `C-c` (to all windows) to replace `kill-server` to achieve this.

Comment (xxchan) [reply Makefile.toml:746]: BTW we have `set -e`, so the `test` after `kill-server` doesn't work..

**Pull request merged**